meshmode/mesh/__init__.py

```diff
@@ -1199,59 +1270,178 @@ def boundary_tag_bit(boundary_tag):
         for i_neighbor_grp in connected_groups:
             is_neighbor_adj = (
                 is_grp_adj & (face_id_pairs[1].groups == i_neighbor_grp))
-            grp_map[i_neighbor_grp] = FacialAdjacencyGroup(
-                igroup=igrp,
-                ineighbor_group=i_neighbor_grp,
-                elements=face_id_pairs[0].elements[is_neighbor_adj],
-                element_faces=face_id_pairs[0].faces[is_neighbor_adj],
-                neighbors=face_id_pairs[1].elements[is_neighbor_adj],
-                neighbor_faces=face_id_pairs[1].faces[is_neighbor_adj])
+            grp_list.append(
+                InteriorAdjacencyGroup(
+                    igroup=igrp,
+                    ineighbor_group=i_neighbor_grp,
+                    elements=face_id_pairs[0].elements[is_neighbor_adj],
+                    element_faces=face_id_pairs[0].faces[is_neighbor_adj],
+                    neighbors=face_id_pairs[1].elements[is_neighbor_adj],
+                    neighbor_faces=face_id_pairs[1].faces[is_neighbor_adj]))
             face_has_neighbor[
                 face_id_pairs[0].faces[is_neighbor_adj],
                 face_id_pairs[0].elements[is_neighbor_adj]] = True
 
         has_bdry = not np.all(face_has_neighbor)
         if has_bdry:
-            faces, elements = np.where(~face_has_neighbor)
-            element_faces = faces.astype(face_id_dtype)
-            neighbors = np.full(len(elements),
-                -(boundary_tag_bit(BTAG_ALL)
-                    | boundary_tag_bit(BTAG_REALLY_ALL)),
-                dtype=element_id_dtype)
+            bdry_indices = np.where(~face_has_neighbor)
+            bdry_element_faces = bdry_indices[0].astype(face_id_dtype)
+            bdry_elements = bdry_indices[1].astype(element_id_dtype)
+            belongs_to_bdry = np.full(
+                (len(boundary_tags), len(bdry_elements)), False)
```

Comment:
The main reason I prefer this is that the unpacking of `np.where(~face_has_neighbor)` is implicitly checking here that we unpack the expected number of items
```suggestion
            bdry_element_faces, bdry_elements = np.where(~face_has_neighbor)
            bdry_element_faces = bdry_element_faces.astype(face_id_dtype)
            bdry_elements = bdry_elements.astype(element_id_dtype)
            belongs_to_bdry = np.full(
                (len(boundary_tags), len(bdry_elements)), False)
```